lib/toml-task/src/lib.rs

```diff
@@ -46,13 +49,29 @@ pub struct Task<T = ordered_toml::Value> {
     pub config: Option<T>,
 
     #[serde(default)]
-    pub interrupts: IndexMap<String, u32>,
+    pub interrupts: IndexMap<String, String>,
     #[serde(default)]
     pub sections: IndexMap<String, String>,
     #[serde(default)]
     pub max_sizes: IndexMap<String, u32>,
 }
 
+impl<T> Task<T> {
+    pub fn notification_bit(&self, name: &str) -> Result<u8> {
+        match self.notifications.iter().position(|n| n == name) {
+            Some(i) => return Ok(i.try_into().unwrap()),
```

Comment:
Sure, done in 1d04a48bc56a9bc3153b93415f98e13bf1f9354e (only checking that it fits in a `u32`, since "does it mask the `userlib::hl` timer notification" isn't really the kernel's problem)